core-common/src/main/java/org/mapstruct/MapperConfig.java

```diff
@@ -80,7 +80,7 @@
      *
```

Comment:
The class-level docs of `@MapperConfig` need an update. AFAIU, only interfaces and abstract classes are valid configuration sources now. Also the comment should mention that any methods declared on the annotated config class are considered prototypes which can be inherited automatically or via `@InheritConfiguration`.